### ISSUE TITLE
Strong naming & GAC support.

### ISSUE THREAD

**Issue by cwensley** [picoe/Eto#408]
This is an issue for discussion.  I am usually not a fan of strong naming because of its various drawbacks (assembly binding redirects, etc).  However, some have asked for this so I'm creating a placeholder to track the issue.

Some thoughts/questions:
- All dependencies need to be strong named, so Eto.Gtk and Eto.[Xam]Mac can't be strong named.
- Does it only make sense to strong name windows platforms? Eto.dll, Eto.WinForms, Eto.Wpf?
- Should the key be stored in github, or is it worth it to keep this private and only for official builds?  This would prevent hackers from creating their own build to steal user information, inject code, etc.

Feedback is welcome!

**Comment by Itaros** [comment]
My thoughts about those:
* I forgot about that... Reasonable point.
* Strongnaming for Mono-powered targets other than Windows should be a thing too. Mono has gacutil for its own GAC magic too.
* The only official source is the repository. Everything else can be maliciously forged regardless. GAC installation is poweruser/admin privilege and having Eto.Forms in the GAC can actually prevent tampering with it on public workstations. At least that is how I see it. There is no harm in keeping key unpublished and only known to owner.

**Comment by cwensley** [comment]
Thanks for the feedback!

As for the GAC/preventing tampering, wouldn't adding Eto.dll to the same folder as your exe skip the GAC?  I guess you could secure the installation folder as well.  Not that security is really the reason for strong naming, but it is related.

**Comment by ArsenShnurkov** [comment]
For *portage* package manager (gentoo, sabayon, calculate, funtoo and other linuxes) it doesn't matter. The packager can sign the assembly (if he wish to) with key from mono (i.e. *mono.snk* file)

**Comment by Itaros** [comment]
Yeah, security is not a reason. But a nice bonus when it comes to assembly selection by identity(strong name) in scenario when deps are deployed as redists. There is, for example, CryptoPro(russian standardized encryption suite) which installs its .net interop into GAC. Nice, convenient. Can be optionally resolved.
